src/platform_impl/web/custom_cursor.rs

```diff
@@ -0,0 +1,56 @@
+use std::sync::Arc;
+
+use wasm_bindgen::{Clamped, JsCast};
+use web_sys::{CanvasRenderingContext2d, HtmlCanvasElement};
+
+use crate::custom_cursor::BadCursor;
+
+#[derive(Debug, Clone)]
+pub struct WebCustomCursor {
+    pub(crate) inner: Arc<str>,
+}
+
+impl WebCustomCursor {
+    pub fn from_rgba(
+        rgba: Vec<u8>,
+        width: u32,
+        height: u32,
+        hotspot_x: u32,
+        hotspot_y: u32,
+    ) -> Result<Self, BadCursor> {
+        #[allow(clippy::disallowed_methods)]
+        let cursor_icon_canvas = web_sys::window()
+            .unwrap()
+            .document()
+            .unwrap()
+            .create_element("canvas")
+            .unwrap()
+            .dyn_into::<HtmlCanvasElement>()
+            .unwrap();
+
+        #[allow(clippy::disallowed_methods)]
+        cursor_icon_canvas.set_width(width);
+        #[allow(clippy::disallowed_methods)]
+        cursor_icon_canvas.set_height(height);
+
+        let context = cursor_icon_canvas
+            .get_context("2d")
+            .unwrap()
+            .unwrap()
+            .dyn_into::<CanvasRenderingContext2d>()
+            .unwrap();
+
+        let image_data =
+            web_sys::ImageData::new_with_u8_clamped_array_and_sh(Clamped(&rgba), width, height);
```

Comment:
This is not supported when the backing memory is a `SharedArrayBuffer`, which is e.g. mandatory when using `atomics`.

See https://github.com/rust-windowing/softbuffer/blob/ab7688e2ed2e2eca51b3c4e1863a5bd7fe85800e/src/web.rs#L196-L223 for some inspiration how to handle this correctly.

Reply:
I'm not sure if we have feature flags like that, so I changed it to always do the extra copies.

Also I don't understand what happens here so I basically copy pasted the reference code.

Reply:
It's just a `target_feature` flag, it has nothing to do with crate features, so it can just work.
I'm afraid having to always do these copies is undesirable.

Let me worry about this, I can fix it later before merging!